docs/docs/resources.md

```diff
@@ -0,0 +1,403 @@
+
+</details>
+
+## A Very Simple Resource
+
+To illustrate the concept, let's create a simple resource that represents the current time.
+
+```js 
+import { cell, resource } from "ember-resources";
+ 
+export const Now = resource(({ on }) => {
+  const now = cell(Date.now());
+ 
+  const timer = setInterval(() => {
+    now.set(Date.now());
+  });
+ 
+  on.cleanup(() => {
+    clearInterval(timer);
+  });
+ 
+  return now;
+});
+```
+
+To see this code in action, [checkout the live demo](https://limber.glimdown.com/edit?c=MQAggiDKAuD2AOB3AhtAxgCwFBbbAJgKYgBmhAdmsdMgOYDOIay5IARsQK72H4hwh8ASwBOhNNAA2ATxBjyREU1gBbeLHIVo9ADRZkjRIUmSQBuRUX8MxPERD1yQ%2BPELRrxSUIButgn5VCRhIRVQA6HAB5eGghDXoALhwAWhAAAy9fNJBUgH15RUYhVnhJZCpcrFSMn2J4MW8hQkRs-MtCESKSsqodEFLy4toPB1hOESozEmgOkcziAo7K6vn%2BhqbEdmNYFpA2hQ6u-p7CPoG0IbnaiwOlDkkdkfoxicJKrDTP2gArRnmsIRqWAidwAbwsz3GvSYxlMAF9SKEVCAAOSEFQcETJMSQ170FEAbhwhAAHuoQcpyPR3AA5R4AXghLyoAAoWeCNCA4QBKED0gB8IFBWBAlOpIHIDJhJhZABFUIQwpLECzudyiSARWL3LFAkpGTxoABJcgzETeZCSNm8gVCrWi5VhQ1yhVKnaq9VankarUaMJoSSEFiceDWvmC4Wi0UBoMiE1mi1W3UdT2i71arViaDjVjKonpnAAHhmajKM35WqN7iEiRAxcBhH5oNBdMQcLhhYA9MmK12SwNyx9PkA&format=glimdown).
+
+<details><summary>In Starbeam</summary>
+
+```js 
+import { Cell, Resource } from "@starbeam/universal";
+ 
+export const Now = Resource(({ on }) => {
+  const now = Cell(Date.now());
+ 
+  const timer = setInterval(() => {
+    now.set(Date.now());
+  });
+ 
+  on.cleanup(() => {
+    clearInterval(timer);
+  });
+ 
+  return now;
+});
+```
+
+</details>
+
+
+> **:bulb:** <br> 
+> A resource's return value is a reactive value. If your resource represents a single cell, it's fine to return it directly. It's also common to return a function which returns reactive data -- that depends on reactive state that you created inside the resource constructor.
+
+When you use the `Now` resource in a component, it will automatically get its lifetime linked to that component. In this case, that means that the interval will be cleaned up when the component is destroyed.
+
+The `resource` function creates a resource Constructor. A resource constructor:
+
+1. Sets up internal reactive state that changes over time.
+2. Sets up the external process that needs to be cleaned up.
+3. Registers the cleanup code that will run when the resource is cleaned up.
+4. Returns a reactive value that represents the current state of the resource as a value.
+
+In this case:
+
+| internal state | external process | cleanup code | return value |
+| ---- | ---- | ---- | ---- |
+| `Cell<number>` | `setInterval` | `clearInterval` | `Cell<number>` |
+
+
+<details><summary>Resource's values are immutable</summary>
+
+When you return a reactive value from a resource, it will always behave like a generic, immutable reactive value. This means that if you return a `Cell` from a resource, the resource's value will have `.current` and `.read()`, but not `.set()`, `.update()` or other cell-specific methods.
```

Comment:
```suggestion
When you return a reactive value from a resource, it will always behave like a generic, immutable reactive value. This means that if you return a `cell` from a resource, the resource's value will have `.current` and `.read()`, but not `.set()`, `.update()` or other cell-specific methods.
```

Looks like this was copied from Starbeam, given the capitalization. Are `current` and `read` still the method names?

Reply:
yeah, `current`, `read`, `set`, and `update` all match.

Thanks for spotting this!